File: Java/KNN.java
import java.util.Arrays;
import java.util.PriorityQueue;

public class KNN {

    public static void main(String[] args) {
        System.out.println(Arrays.deepEquals(kClosest(new int[][] {{ 1, 3 }, { -2, 2 }}, 1), 
        new int[][] {{-2, 2}}));
        System.out.println(Arrays.deepEquals(kClosest(new int[][] {{ 3, 3 }, { 5, -5 }, { -2, 4}}, 2), 
        new int[][] {{ 3, 3 }, { -2, 4}}));
    }

    public static int[][] kClosest(int[][] points, int K) {
        PriorityQueue<int[]> pq = new PriorityQueue<int[]>((p1, p2) -> p2[0] * p2[0] + p2[1] * p2[1] - p1[0] * p1[0] - p1[1] * p1[1]);
        for (int[] p : points) {
            pq.offer(p);
            if (pq.size() > K) {
                pq.poll();
            }
        }
        int[][] res = new int[K][2];
        while (K > 0) {
            res[--K] = pq.poll();
        }
        return res;
    }
}

File: Python/KNN.py
import heapq

def kClosest(points, K):
    max_heap = []
    for p in points:
        dist = p[0] * p[0] + p[1] * p[1]
        heapq.heappush(max_heap, (-dist, p))
        if len(max_heap) > K:
            heapq.heappop(max_heap)
    res = []
    while K > 0:
        res.append(heapq.heappop(max_heap)[1])
        K -= 1
    return res

print(kClosest([[1, 3], [-2, 2]], 1) == [[-2, 2]])
print(kClosest([[3, 3], [5, -5], [-2, 4]], 2) ==  [[-2, 4], [3, 3]])